src/v2/Apps/Conversation/Components/Conversation.tsx

```diff
@@ -179,36 +179,38 @@ const Conversation: React.FC<ConversationProps> = props => {
                   style={{ alignSelf: "center" }}
                   time={messageGroup[0].createdAt}
                   exact
-                  mt={0.5}
                   mb={1}
                 />
                 {messageGroup.map((message, messageIndex) => {
                   const nextMessage = messageGroup[messageIndex + 1]
+                  const senderChanges =
+                    nextMessage && nextMessage.isFromUser !== message.isFromUser
+                  const lastMessageInGroup =
+                    messageIndex === messageGroup.length - 1
+                  const spaceAfter =
+                    senderChanges || lastMessageInGroup ? 2 : 0.5
 
                   return (
-                    <Message
-                      message={message}
-                      initialMessage={conversation.initialMessage}
-                      key={message.internalID}
-                      isFirst={groupIndex + messageIndex === 0}
-                      showTimeSince={
-                        message.createdAt &&
-                        today &&
-                        messageGroup.length - 1 === messageIndex
-                      }
-                      mb={
-                        nextMessage &&
-                        nextMessage.isFromUser !== message.isFromUser
-                          ? 1
-                          : undefined
-                      }
-                    />
+                    <>
+                      <Message
+                        message={message}
+                        initialMessage={conversation.initialMessage}
+                        key={message.internalID}
+                        isFirst={groupIndex + messageIndex === 0}
+                        showTimeSince={
+                          message.createdAt &&
+                          today &&
+                          messageGroup.length - 1 === messageIndex
+                        }
+                      />
+                      <Spacer mb={spaceAfter} />
```

Comment:
Tried to move the padding between messages in one place and have it in `spaceAfter`. So the space between messages is `20px` when there is a new message group or the sender changes, otherwise, it is `5px`

Reply:
Nice solution to some complex behavior!